packages/server/__tests__/db/db.test.js

```diff
@@ -247,4 +248,107 @@ describe('db', () => {
+                    status: emailConstants.emailSubscriptionStatus.subscribed,
+                },
+                {
+                    agency_id: fixtures.agencies.accountancy.id,
+                    notification_type: emailConstants.notificationType.grantInterest,
+                    status: emailConstants.emailSubscriptionStatus.subscribed,
+                },
+            ];
+            await knex('email_subscriptions').insert(preferences);
+            const result = await db.getAgencyEmailSubscriptionPreference(fixtures.agencies.accountancy.id);
+            expect(result).to.have.all.keys(...Object.values(emailConstants.notificationType));
+            const resultSet = new Set(Object.values(result));
+            expect(resultSet.size).to.equal(1);
+            expect(resultSet.has(emailConstants.emailSubscriptionStatus.subscribed)).to.equal(true);
+        });
+        it('sets default email subscription preferences if none exist', async () => {
+            await db.setAgencyEmailSubscriptionPreference(fixtures.agencies.accountancy.id, {});
+            const result = await db.getAgencyEmailSubscriptionPreference(fixtures.agencies.accountancy.id);
+            expect(result).to.have.all.keys(...Object.values(emailConstants.notificationType));
+            const resultSet = new Set(Object.values(result));
+            expect(resultSet.size).to.equal(1);
+            expect(resultSet.has(emailConstants.emailSubscriptionStatus.unsubscribed)).to.equal(true);
+        });
+        it('sets custom email subscription preferences', async () => {
+            await db.setAgencyEmailSubscriptionPreference(
+                fixtures.agencies.accountancy.id,
+                {
+                    [emailConstants.notificationType.grantAssignment]: emailConstants.emailSubscriptionStatus.subscribed,
+                },
+            );
+            const result = await db.getAgencyEmailSubscriptionPreference(fixtures.agencies.accountancy.id);
+            expect(result).to.have.all.keys(...Object.values(emailConstants.notificationType));
+            expect(result[emailConstants.notificationType.grantAssignment]).to.equal(emailConstants.emailSubscriptionStatus.subscribed);
+            const resultSet = new Set(Object.values(result));
+            expect(resultSet.size).to.equal(2);
+            expect(resultSet.has(emailConstants.emailSubscriptionStatus.unsubscribed)).to.equal(true);
+            expect(resultSet.has(emailConstants.emailSubscriptionStatus.subscribed)).to.equal(true);
+        });
+        it('raises error if unknown notification type is supplied', async () => {
+            let expectedError = '';
+
+            const test = async () => {
+                try {
+                    await db.setAgencyEmailSubscriptionPreference(fixtures.agencies.accountancy.id, {
+                        UNKNOWN: emailConstants.emailSubscriptionStatus.subscribed,
```

Comment:
Where is UNKNOWN defined? I can't find it anywhere. :(

Reply:
It isn't defined anywhere, I just made it up in this line to make sure that when we pass an unrecognized/unknown value to the function it will raise an error.